test/com/google/enterprise/adaptor/ad/AdAdaptorTest.java

```diff
@@ -1425,6 +1419,44 @@ public void testFakeAdaptorGetDocIds() throws Exception {
   }
 
   @Test
+  public void testMultipleFullGroupPushes() throws Exception {
+    MockLdapContext ldapContext = defaultMockLdapContext();
+    AdAdaptor adAdaptor = new FakeAdaptor(ldapContext);
+    RecordingDocIdPusher pusher = new RecordingDocIdPusher();
+    Map<String, String> configEntries = defaultConfig();
+    pushGroupDefinitions(adAdaptor, configEntries, pusher, /*fullPush=*/ true,
+        /*init=*/ true);
+
+    HashMap<GroupPrincipal, Collection<Principal>> goldenGroups =
+        new HashMap<GroupPrincipal, Collection<Principal>>();
+    Principal everyone = new GroupPrincipal("Everyone", "Default");
+    goldenGroups.put((GroupPrincipal) everyone, new ArrayList<Principal>());
+    goldenGroups.put(new GroupPrincipal("Authenticated Users@NT Authority",
+        "Default"), Arrays.asList(everyone));
+    goldenGroups.put(new GroupPrincipal("Interactive@NT Authority",
+        "Default"), Arrays.asList(everyone));
+    assertEquals(goldenGroups, pusher.getGroupDefinitions());
+
+    // Add the sam group.
+    addDefsToMockLdapContext(ldapContext, false);
+    pushGroupDefinitions(adAdaptor, configEntries, pusher, /*fullPush=*/ true,
+        /*init=*/ false);
+    @SuppressWarnings("unchecked")
+    HashMap<GroupPrincipal, Collection<Principal>> newGoldenGroups =
+        (HashMap<GroupPrincipal, Collection<Principal>>) goldenGroups.clone();
+    newGoldenGroups.put(new GroupPrincipal("sam@GSA-CONNECTORS", "Default"),
+        new ArrayList<Principal>());
+    assertEquals(newGoldenGroups, pusher.getGroupDefinitions());
+
+    // Clear the added group.
+    ldapContext.clearSearchResults();
```

Comment:
Why does clearing the search results from the context still return the original groups?

Reply:
There are default, built-in groups that always get returned. Yes, I expected an empty group map as well at first.